### ISSUE TITLE
Add predict_proba() for LIME, SHAP, and ONNX compatibility

### ISSUE THREAD

**Issue by direkkakkar319-ops** [skeval-ai/skeval#41]
## Summary
Adding `predict_proba()` unlocks compatibility with LIME, SHAP, and ONNX — three widely used ML libraries that require confidence scores per class rather than just a predicted label.

## What predict_proba() returns
Instead of a single predicted label, it returns a probability for every class:

```python
clf.predict_proba(["The sky is blue"])
# [[0.91, 0.04, 0.03, 0.02]]  — shape (n_samples, n_classes)
```

The output follows the sklearn convention: a 2D numpy array of shape `(n_samples, n_classes)`, where columns are ordered by `clf.label_encoder.idx2label`.

## Changes required
- Add `predict_proba(X: List[str]) -> np.ndarray` to `SentenceClassifier`
- Apply `torch.softmax()` to the model logits to convert raw scores to probabilities
- Add a `classes_` attribute on `SentenceClassifier` (set during `fit()`) — required by sklearn convention
- Raise `RuntimeError` if called before `fit()` or `load()`

## Implementation sketch
```python
def predict_proba(self, X: List[str]) -> np.ndarray:
    if self.model is None:
        raise RuntimeError("Model is not fitted. Call fit() or load() first.")
    self.model.eval()
    results = []
    with torch.no_grad():
        for sentence in X:
            encoded = torch.tensor(self.vocab.encode(sentence), dtype=torch.long)
            if len(encoded) == 0:
                encoded = torch.tensor([0], dtype=torch.long)
            output = self.model(encoded.to(self.device),
                                torch.tensor([0]).to(self.device))
            probs = torch.softmax(output, dim=1).cpu().numpy()
            results.append(probs[0])
    return np.array(results)
```

## Libraries this unlocks
| Library | Use case |
|---|---|
| LIME | Explain individual sentence predictions |
| SHAP | Feature importance per prediction |
| ONNX (sklearn-onnx) | Export model for production inference outside Python |

## Acceptance criteria
- `predict_proba()` returns a 2D numpy array of shape `(n_samples, n_classes)`
- Each row sums to 1.0
- `classes_` attribute is set on the classifier after `fit()`
- LIME and SHAP work with the output without any extra wrapping

**Milestone:** v0.2.0
